### DISCUSSION TITLE
How do I use `Interaction.Message` for slash commands?

### DISCUSSION THREAD

**Discussion by XangelMusic** [bwmarrin/discordgo#1487] [Q&A] [answered]
I want to specifically get the URL of an image attachment from a message, and I want to interact with message with a menu slash command.

The problem is however `Interaction.Message` doesn't work; the Message object is always nil. I'm also baffled by the comment "this field is only filled when a button click triggered the interaction. Otherwise it will be nil." But I am clicking the command from the right click menu on the message.

What am I doing wrong?

**Comment by Nv7-GitHub** [comment]
Do you have the message intent

**Reply by XangelMusic** [reply]
I don't think so. Can you explain to me about message intents please?

**Reply by Nv7-GitHub** [reply]
https://support-dev.discord.com/hc/en-us/articles/4404772028055-Message-Content-Privileged-Intent-FAQ

**Reply by XangelMusic** [reply]
Okay, I have looked into the intent and Message Intents was already enabled when I checked the developer portal. So I don't think its message intents that is the issue.

**Comment by XangelMusic** [comment, marked as answer]
I found the answer. I was supposed to look at the Context Menus example to make a context menu command.
